### ISSUE TITLE
Custom Validator Errors are not logged

### ISSUE THREAD

**Issue by eschwartz** [mattallty/Caporal.js#94]
I notice that if I write a custom function validator, any errors thrown by my validator are not logged. Instead I just get a generic error like:

```
Error: Invalid value 'bar' for option --foo
```

This is a bummer, because I would like to notify the user as to _why_ their argument/option is invalid.

For example, I wrote a `DateValidator`, to allow the users to pass in dates to my command:

```
function DateValidator(opt) {
	const date = new Date(opt);

	if (date.toString() === 'Invalid Date') {
		throw new Error(`"${opt}" is not a valid date. Option value must be an ISO date string or a timestamp in milliseconds`);
	}

	return date;
}
```

I would really like to be able to show that validation error to the user, so they don't have to guess as to why their command failed. For example:

```
Error: Invalid value 'today' for option --start-time
  "today" is not a valid date. Option value must be an ISO date string or a timestamp in milliseconds
```

To work around this, I overwrote the `fatalError` method of Caporal:

```js
const prog = require('caporal');
const fatalErrorOrig = prog.fatalError.bind(cli);
prog.fatalError = (err) => {
	while (err.meta && err.meta.originalError) {
		err = err.meta.originalError;
	}

	fatalErrorOrig(err);
};
```

This, of course, is not ideal.

I'd be willing to put together a pull request for this, if you're open to it.

**Comment by eschwartz** [comment]
Also see #51

**Comment by mattallty** [comment]
fixed in #98

**Comment by ramtinsoltani** [comment]
I'm actually having the same problem, even though it's claimed to be fixed in #98 
Can we reopen this issue?

I'm using Caporal 1.3.0 on macos 10.15.2 with node 12.13.1

Here's my code:

```ts
import program from 'caporal';

program.
  .command('test', 'Test command')
  .argument('<arg>', 'Argument test', arg => {
    if ( arg === '1' ) throw new Error('Custom error...');
    return arg;
  })
  .action(() => {});

program.parse(process.argv);
```

Running `whatever test 1` will print:
```
   Error: Invalid value '1' for argument arg.
   undefined
```

BTW, what causes `undefined` to be displayed under the error message?
